src/Arcus.Security.Providers.AzureKeyVault/Authentication/ServicePrincipalAuthentication.cs

```diff
@@ -37,19 +55,34 @@ public ServicePrincipalAuthentication(string clientId, string clientKey)
         /// <exception cref="InvalidOperationException">When the JSON web token (JWT) cannot be obtained.</exception>
         public Task<IKeyVaultClient> AuthenticateAsync()
         {
+            _logger.LogTrace("Start authenticating with service principal to Azure Key Vault...");
             IKeyVaultClient keyVaultClient = new KeyVaultClient(GetTokenAsync);
+            _logger.LogInformation("Authenticated with service principal to Azure Key Vault");
+
             return Task.FromResult(keyVaultClient);
         }
 
         private async Task<string> GetTokenAsync(string authority, string resource, string scope)
```

Comment:
See other comment, this one certainly has a lot of overlap with src/Arcus.Security.Providers.AzureKeyVault/Authentication/CertificateBasedAuthentication.cs